apps/wallet/src/ui/app/pages/accounts/AddAccountPage.tsx

```diff
@@ -2,86 +2,111 @@
 // SPDX-License-Identifier: Apache-2.0
 
 import { LedgerLogo17 as LedgerLogo } from '@mysten/icons';
-import { useState, type ReactNode } from 'react';
+import { useState, type ReactNode, useEffect, useRef, useCallback } from 'react';
 import toast from 'react-hot-toast';
 import { useNavigate, useSearchParams } from 'react-router-dom';
 import Browser from 'webextension-polyfill';
 import { useAccountsFormContext } from '../../components/accounts/AccountsFormContext';
+import { ZkLoginButtons } from '../../components/accounts/ZkLoginButtons';
 import { ConnectLedgerModal } from '../../components/ledger/ConnectLedgerModal';
 import { getLedgerConnectionErrorMessage } from '../../helpers/errorMessages';
 import { useAppSelector } from '../../hooks';
+import { useCountAccountsByType } from '../../hooks/useCountAccountByType';
+import { useCreateAccountsMutation } from '../../hooks/useCreateAccountMutation';
 import { AppType } from '../../redux/slices/app/AppType';
-import { SocialButton } from '../../shared/SocialButton';
 import { Button } from '_app/shared/ButtonUI';
 import { Text } from '_app/shared/text';
 import Overlay from '_components/overlay';
+import { type ZkProvider, zkProviderDataMap } from '_src/background/accounts/zk/providers';
 import { ampli } from '_src/shared/analytics/ampli';
 
+async function openTabWithSearchParam(searchParam: string, searchParamValue: string) {
+	const { origin, pathname, hash } = window.location;
+	const [currentHash, currentHashSearch] = hash.split('?');
+	const urlSearchParams = new URLSearchParams(currentHashSearch);
+	urlSearchParams.set(searchParam, searchParamValue);
```

Comment:
nit: can tweak this slightly to avoid manually constructing the url
```typescript
async function openTabWithSearchParam(searchParam: string, searchParamValue: string) {
    const currentUrl = new URL(window.location);
    const [currentHash, currentHashSearch] = currentUrl.hash.split('?');
    const urlSearchParams = new URLSearchParams(currentHashSearch);
    urlSearchParams.set(searchParam, searchParamValue);
    currentUrl.hash = currentHash + '?' + urlSearchParams.toString();
    
    await Browser.tabs.create({
        url: currentUrl.href,
    });
}

```

Reply:
Updated